src/spawn.c

```diff
@@ -0,0 +1,1121 @@
+
+		for (s = command_line; !strchr(BLANKS, *s); s++);
+
+		if (quote && quote < s)
+		{
+			g_set_error(error, G_SHELL_ERROR, G_SHELL_ERROR_BAD_QUOTING,
+				_("A quoted Windows program name must be entirely inside the quotes"));
+			return FALSE;
+		}
+	}
+
+	program = g_strndup(command_line, s - command_line);
+	arguments = s + (*s == '"');
+
+	for (s = arguments; *s; s++)
+	{
+		if (*s == '"')
+		{
+			const char *slash;
+
+			for (slash = s; slash > arguments && slash[-1] == '\\'; slash--);
+			if ((s - slash) % 2 == 0)
+				open_quote ^= TRUE;
+		}
+	}
+
+	if (open_quote)
+	{
+		/* from glib, except for the fixed quote */
+		g_set_error(error, G_SHELL_ERROR, G_SHELL_ERROR_BAD_QUOTING,
+			_("Text ended before matching quote was found for %c."
+			  " (The text was '%s')"), '"', command_line);
+		return FALSE;
+	}
+#else  /* G_OS_WIN32 */
+	int argc;
+	char **argv;
+
+	if (!g_shell_parse_argv(command_line, &argc, &argv, error))
+		return FALSE;
+
+	program = g_strdup(argv[0]);  /* note: empty string results in parse error */
+	g_strfreev(argv);
+#endif  /* G_OS_WIN32 */
+
+	if (execute && !g_find_program_in_path(program))
+	{
+		g_set_error(error, G_SHELL_ERROR, G_SHELL_ERROR_FAILED,  /* or SPAWN error? */
+			_("Program '%s' not found in path"), program);
+		g_free(program);
+		return FALSE;
+	}
+
+	g_free(program);
+	return TRUE;
+}
+
+
+/**
+ *  Kills a process.
+ *
+ *  @param pid id of the process to kill.
+ *  @param error return location for error.
+ *
+ *  @return @c TRUE on success, @c FALSE if an error was set.
+ **/
+gboolean spawn_kill_process(GPid pid, GError **error)
+{
+#ifdef G_OS_WIN32
+	if (!TerminateProcess(pid, 15))
+	{
+		gchar *message = g_win32_error_message(GetLastError());
+
+		g_set_error(error, G_SPAWN_ERROR, G_SPAWN_ERROR_FAILED,
+			_("TerminateProcess() failed: %s"), message);
+		g_free(message);
+		return FALSE;
+	}
+#else
+	if (kill(pid, SIGTERM))
+	{
+		g_set_error(error, G_SPAWN_ERROR, G_SPAWN_ERROR_FAILED, g_strerror(errno));
+		return FALSE;
+	}
+#endif
+	return TRUE;
+}
+
+
+#ifdef G_OS_WIN32
+static gchar *spawn_create_process_with_pipes(char *command_line, const char *working_directory,
+	void *environment, HANDLE *hprocess, int *stdin_fd, int *stdout_fd, int *stderr_fd)
+{
+	enum { WRITE_STDIN, READ_STDOUT, READ_STDERR, READ_STDIN, WRITE_STDOUT, WRITE_STDERR };
+	CRITICAL_SECTION critical;  /* probably unneeded, but good practice */
+	STARTUPINFO startup;
+	PROCESS_INFORMATION process;
+	HANDLE hpipe[6] = { NULL, NULL, NULL, NULL, NULL, NULL };
+	int *fd[3] = { stdin_fd, stdout_fd, stderr_fd };
+	const char *failed;     /* failed WIN32/CRTL function, if any */
+	gchar *message = NULL;  /* glib WIN32/CTRL error message */
+	gchar *failure = NULL;  /* full error text */
+	int i;
+
+	ZeroMemory(&startup, sizeof startup);
+	startup.cb = sizeof(startup);
+	startup.dwFlags = STARTF_USESHOWWINDOW;
+	startup.wShowWindow = SW_SHOWMINNOACTIVE;  /* so you can switch to it if needed */

```

Comment:
Generally not, but if you don't capture the stdout/stderr, you'd better have some way to see it. May be useful for GUI tools, though it should not minimize the window then. In both these cases, g_spawn_[a]sync should suffice.